### DISCUSSION TITLE
AD only log in

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2609] [Ideas]
Hi Team, 
I think a good addition to BB would be to have a toggle to eliminate the regular log-in when AD is activated to help not confuse people.  a bit more streamlined. 

![image](https://user-images.githubusercontent.com/80058238/133347066-4f0cb9ee-3abd-4aed-90f9-fb969c443698.png)

**Comment by joebudi** [comment]
This is a very fair suggestion. I think there is a project that focuses on UX for the end-user. This would also include:
#2605

**Reply by aderici** [reply]
I really like how the op marked that area with a red cross. Thats how i'm feeling right now :) need this desperately for a project.

**Comment by jonnymccullagh** [comment]
Maybe needs a small link somewhere on the page for email login so the admin can bypass the auth source if the auth source is down or wrongly configured?

**Reply by aderici** [reply]
Lets dig deeper into the root cause / requirement here

There are admins, and there are app users and there are sometimes even external organizations (think about a support portal where you want to provide access for your customers)

Lets say internal users will use google auth, you dont want them to be able to use password login. Because you dont want them to have another set of passwords. Because when they leave the company you just want to terminate their google and thats it. If they have a password they will still be able to login. Yet another credential to worry about.

So its not only hiding the user pass fields (thats for cosmetics), its actually not allowing users to login with password even if the fields were there. You could always allow the main admin account to use password or pick per user

**Comment by R2bEEaton** [comment]
This is an incredible idea! I have already had several people call our internal IT support because they try to use their credentials in the boxes instead using of the AD button. These are people that were directly told and shown in person how to log in, and had been doing it for a couple of weeks with no problem.

Like you say, sometimes people forget and get confused - I think this would be a great solution to get it a little more streamlined for the end user. It could be a good addition to the sparse Organisation or Theming Settings page.

**Comment by martkaczmarek** [comment]
For anyone looking for a temporary solution: I just add users accounts with randomly generated password and I edited the Reset Password email template - now the template doesn't contain the new password, and also tells the user to just log in with SSO.
Technically they may try to log in with email, but they don't know the password.

**Reply by aderici** [reply]
I ended up modifying source codes, copying login page to an alternative name and modifying the original to remove user password login.

**Reply by mjashanks** [reply]
Hey @aderici I'd just like to point out that Budibase is GPL licensed. Any code change that you make should be public and open-source :)

Thanks.

Would definitely like to have a solution for this though.

**Reply by aderici** [reply]
@mjashanks i think the change i made is self explanatory.

**Reply by mjashanks** [reply]
Great - looks like a good simple solution. Therefore, It should be easy to commit and push them to a public GitHub repository for everyone to benefit from.

We have very purposefully chosen the GPL license so that any changes to the codebase are made public and open-source under the GPL. Please respect our community by sharing your changes.

**Reply by aderici** [reply]
@mjashanks something like this? https://github.com/aderici/budibase-mod

**Reply by mjashanks** [reply]
@aderici I appreciate that, thank you :)
